homeassistant/components/deconz/climate.py

```diff
@@ -64,11 +62,11 @@
 }
 DECONZ_TO_FAN_MODE = {value: key for key, value in FAN_MODE_TO_DECONZ.items()}
 
-HVAC_MODE_TO_DECONZ: dict[str, str] = {
-    HVAC_MODE_AUTO: THERMOSTAT_MODE_AUTO,
-    HVAC_MODE_COOL: THERMOSTAT_MODE_COOL,
-    HVAC_MODE_HEAT: THERMOSTAT_MODE_HEAT,
-    HVAC_MODE_OFF: THERMOSTAT_MODE_OFF,
+HVAC_MODE_TO_DECONZ: dict[HVACMode | str, str] = {
```

Comment:
I think it should only be HVACMode as a type here
```suggestion
HVAC_MODE_TO_DECONZ: dict[HVACMode, str] = {
```

Reply:
I have adjusted it, although I had to add some `type:ignore` statements.
I think these ignore statements will simply becoe redundant when the platform is updated to drop support for `str`.